cloud/pkg/dynamiccontroller/application/application.go

```diff
@@ -341,22 +341,20 @@ func NewApplicationCenter(dynamicSharedInformerFactory dynamicinformer.DynamicSh
 	return a
 }
 
-func toBytes(i interface{}) []byte {
+func toBytes(i interface{}) (bytes []byte) {
 	if i == nil {
-		return []byte{}
+		return
+	}
+
+	if bytes, ok := i.([]byte); ok {
+		return bytes
 	}
-	var bytes []byte
+
 	var err error
-	switch i := i.(type) {
-	case []byte:
-		bytes = i
-	default:
-		bytes, err = json.Marshal(i)
-		if err != nil {
-			klog.Fatalf("marshal content to []byte failed, err: %v", err)
-		}
+	if bytes, err = json.Marshal(i); err != nil {
+		klog.Errorf("marshal content to []byte failed, err: %v", err)
 	}
-	return bytes
+	return
```

Comment:
lgtm, the return here maybe no needed?

Reply:
In golang, if there is a return value, then `return` is required.

Reply:
Okey, get it.
/approve